### PR TITLE
Implement Thesis Detail Page

### PR DESCRIPTION
**Pull request by fabian-emilius**
This PR implements basic functionality of the Thesis page (https://github.com/ls1intum/thesis-track/issues/74)

Testing url: https://thesis-track-dev.ase.cit.tum.de/theses

Features
- Allow advisors and supervisors to create a thesis on the thesis overview page
- Show submitted applications and your thesis on the dashboard page
- Show all existing thesis on the thesis overview page
- Allow advisors to configure a thesis (title, visibility, start date, end date, assigned users, state change dates)
- Allow advisors to close a thesis if student dropped out
- Allow students to set abstract and an info text for a thesis
- Allow students to upload a proposal
- Allow advisors to accept a proposal
- Allow students to upload a thesis
- Allow students to upload a presentation
- Allow students to mark submission as final
- Allow advisors to write an assessment for the final thesis
- Allow supervisors to grade and complete the thesis

No deployment changes are needed for this PR

### CODE REVIEW

File: client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx [hunk 1]
```diff
@@ -1,34 +1,57 @@
...
+        defaultStates={
+          managementAccess
+            ? [
+                ThesisState.PROPOSAL,
+                ThesisState.WRITING,
+                ThesisState.SUBMITTED,
+                ThesisState.ASSESSED,
+                ThesisState.GRADED,
+              ]
+            : [ThesisState.FINISHED]
```
Comment (bgeisb) [inline client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx:38]:
```suggestion
        defaultStates={}
```
Wouldn't it make sense to see all theses here by default?

General Question: 
How is this page relevant for students? While it’s true that other theses can be shared with students, introducing a new tab for this purpose might make sense due to its distinct use case.

File: client/src/pages/ThesisPage/ThesisPage.tsx [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
-    <>
-      <ThesisConfigSection />
-      <ThesisInfoSection />
-      <ThesisProposalSection />
-      <ThesisWritingSection />
-      <ThesisAssessmentSection />
-      <ThesisFinalGradeSection />
-    </>
+    <ThesisProvider thesisId={thesisId} requireLoadedThesis>
+      <ContentContainer>
```
Comment (bgeisb) [inline client/src/pages/ThesisPage/ThesisPage.tsx:21]: (Optional)

```suggestion
      <ContentContainer style={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
```

File: client/src/components/ThesesGanttChart/ThesesGanttChart.tsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const ThesesGanttChart = () => {
+  return <></>
```
Comment (bgeisb) [inline client/src/components/ThesesGanttChart/ThesesGanttChart.tsx:2]: Can this file be deleted?

**Review by bgeisb — CHANGES_REQUESTED** [review]

File: client/src/components/ThesesGanttChart/ThesesGanttChart.tsx (hunk 3 above)
Comment (fabian-emilius) [reply client/src/components/ThesesGanttChart/ThesesGanttChart.tsx:2]: Right now this is a placeholder file for a future PR, so it will be implemented eventually

File: client/src/pages/ThesisPage/ThesisPage.tsx (hunk 2 above)
Comment (fabian-emilius) [reply client/src/pages/ThesisPage/ThesisPage.tsx:21]: What would be the benefit of that (just curious)

File: client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx (hunk 1 above)
Comment (fabian-emilius) [reply client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx:38]: For advisors and supervisors this page will be used to check the progress and have an overview of all ongoing theses (this is a requirement from Stephan)

While implementing that, I thought why not make this page also available to everyone, especially since theses have a visibility setting (PRIVATE, INTERNAL, PUBLIC) which configures who can access it. Could be cool for students to browse finished theses that are marked as public.

File: client/src/components/ApplicationReviewModal/ApplicationReviewModal.tsx
[the remark this answers is not in the document]
Comment (fabian-emilius) [reply client/src/components/ApplicationReviewModal/ApplicationReviewModal.tsx:1]: This wasnt really part of this PR. I just moved the code around to improve reusability. Overall I agree with all of your points but its still part of the legacy application review process that I will rebuild so I would not want to change it now since I will build a new page for that anyway

File: client/src/components/ApplicationReviewForm/ApplicationReviewForm.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            placeholder='Thesis Title'
+            label='Thesis Title'
+            {...form.getInputProps('title')}
+          />
+
+          <UserMultiSelect
+            label='Supervisor'
+            required={true}
+            groups={['supervisor']}
+            maxValues={1}
```
[the remark this answers is not in the document]
Comment (fabian-emilius) [reply client/src/components/ApplicationReviewForm/ApplicationReviewForm.tsx:96]: Its a requirement from Stephan to only have 1 supervisor selectable. Without this line multiple would be supported so it can be changed easily when requirements would change

File: client/src/app/layout/AuthenticatedArea/AuthenticatedArea.tsx
[the remark this answers is not in the document]
Comment (fabian-emilius) [reply client/src/app/layout/AuthenticatedArea/AuthenticatedArea.tsx:1]: `/management/thesis-applications` currently routes to the legacy application review process that I plan to rebuild on `/applications`. This route and feature already existed when I started working on thesis track so its mainly for url compatibility until I implement the new process

File: client/src/app/layout/AuthenticatedArea/AuthenticatedArea.tsx [hunk 5]
```diff
@@ -56,16 +56,15 @@ const links: Array<{
     label: 'Create Topic',
     icon: FolderSimplePlus,
     roles: ['admin', 'advisor', 'supervisor'],
+  },*/
+  {
+    link: '/theses',
+    label: 'Theses Overview',
+    icon: Kanban,
+    roles: undefined,
```
[the remark this answers is not in the document]
Comment (fabian-emilius) [reply client/src/app/layout/AuthenticatedArea/AuthenticatedArea.tsx:64]: Answered in the discussion on ThesisOverviewPage

File: client/src/app/Buttons.module.css
[the remark this answers is not in the document]
Comment (fabian-emilius) [reply client/src/app/Buttons.module.css:1]: Its in this folder because its only used in App.tsx to configure new button variants. I agree that its a bit out of place and I will move it to a separate folder

File: client/src/pages/ThesisPage/ThesisPage.tsx (hunk 2 above)
Comment (bgeisb) [reply client/src/pages/ThesisPage/ThesisPage.tsx:21]: You would get rid of all the spacers and can achieve the same behavior with one line of code. Still in the end it's just personal taste I guess. :)

File: client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx (hunk 1 above)
Comment (bgeisb) [reply client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx:38]: I see, I thought this page is meant to be something like a hub for supervisors and advisors. 

Maybe a more intuitive name for this tab would be a good enhancement. 
Something like "Browse Theses" or "All Theses"/"Active Theses"...still - I don't want to overrule any of @krusche's requirements.

Either way, from a usability perspective I would leave the filters empty by default.

File: client/src/components/ThesesGanttChart/ThesesGanttChart.tsx (hunk 3 above)
Comment (bgeisb) [reply client/src/components/ThesesGanttChart/ThesesGanttChart.tsx:2]: Ah, I see. 
If this has nothing to do with this PR I would keep it as a local change next time.

File: client/src/components/ApplicationReviewModal/ApplicationReviewModal.tsx
[the remark this answers is not in the document]
Comment (bgeisb) [reply client/src/components/ApplicationReviewModal/ApplicationReviewModal.tsx:1]: Sure. I just noticed that during the review and thought I'd let you know. Feel free to change this in a future PR. :)

**Review by bgeisb — CHANGES_REQUESTED** [review]
LGTM 🙂👍

The only thing left to address are the default filters. The other changes can be handled in future PRs.

As we discussed on Slack, it would be really helpful if future PRs could be smaller and more self-contained. Thank you! 🚀

File: client/src/components/ApplicationReviewModal/ApplicationReviewModal.tsx
[the remark this answers is not in the document]
Comment (fabian-emilius) [reply client/src/components/ApplicationReviewModal/ApplicationReviewModal.tsx:1]: Yes I will. Thanks for the suggestions

File: client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx (hunk 1 above)
Comment (fabian-emilius) [reply client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx:38]: I have changed the page to be only visible for advisors and supervisors again as it was requested by Stephan as well.
There also was a discussion about the default states but we agreed to discuss this once the Gantt chart on that page is also implemented as we do not have the full picture right now

File: client/src/components/ThesesGanttChart/ThesesGanttChart.tsx (hunk 3 above)
Comment (fabian-emilius) [reply client/src/components/ThesesGanttChart/ThesesGanttChart.tsx:2]: yeah I agree. Will do that next time

File: client/src/pages/ThesisPage/ThesisPage.tsx (hunk 2 above)
Comment (fabian-emilius) [reply client/src/pages/ThesisPage/ThesisPage.tsx:21]: It did not really work as it changed also gaps inside the accordions. I would leave it like this for now and come back to that at a later point

**Review by bgeisb — APPROVED** [review]
